api/apps/api/src/modules/projects/projects-crud.service.ts

```diff
@@ -199,4 +213,51 @@ export class ProjectsCrudService extends AppBaseService<
       return this.planningUnitsService.create(createModel);
     }
   }
+
+  async setDataUpdate(
+    model: Project,
+    update: UpdateProjectDTO,
+    _?: AppInfoDTO,
+  ): Promise<Project> {
+    const geometry = await this.resolveBBoxFromAdminArea(update, _);
+    if (geometry?.bbox) {
+      const modelWithBbox = await super.setDataUpdate(model, update, _);
+      modelWithBbox.bbox = geometry.bbox;
+      return modelWithBbox;
+    }
+    return model;
+  }
+
+  private async resolveBBoxFromAdminArea(
+    model: CreateProjectDTO | UpdateProjectDTO,
+    _?: AppInfoDTO,
+  ): Promise<Polygon | undefined> {
+    if (!this.bboxResolver.shouldResolveBbox(model)) {
+      return;
+    }
+    const derivedSubmittedAdminArea = await this.getPlanningArea({
+      ...model,
+      extent: undefined,
+    });
+
+    let geometry;
+
+    try {
+      geometry = this.bboxResolver.resolveBBox(
+        model,
+        derivedSubmittedAdminArea,
+      );
+    } catch (error) {
+      // currently no easy way to propagate specific error to controller, thus throwing Http error already
+      throw new BadRequestException(error);
+    }
+
+    if (geometry) {
+      return geometry;
+    } else {
+      throw new InternalServerErrorException(
+        `Missing bbox for given admin region.`,
```

Comment:
not a big deal, but since we already have the planning area, it could be useful to add country/l1/l2 (where they are defined) to the error message - that we have no bbox should normally not happen, but when looking at log traces in staging/prod, this info can help get to the right bit of (incorrect) data easily.

Reply:
@hotzevzl correct me if I am wrong but if it is Http Error, all the payload (assuming redact of some sensitive data) should be there already - they are usually captured by monitoring services (like Sentry). What benefit I am missing?

Reply:
you are right - I was thinking about our current `AllExceptionsFilter`, where we do log some error data, but not the whole request or response (or relevant parts of them at least) - indeed, adding at least request payload to what we log (maybe behind a feature flag) could help to provide better context to our raw logs (even without using a monitoring service).

Let's leave this as is then - by logging request payloads we could get context for all errors rather than having to care about adding some context to individual exception being thrown.